### PR TITLE
Reorder Android setup docs

### PR DESCRIPTION
**Pull request by Voklen**
## Changes

When I was trying to run the example project it worked perfectly on Linux but for Android I had to first setup the NDK. I've therefore put that bit in the initial tutorial for running for Android and then linked to it from the template Android setup. I've also move the `Alternative NDK setup` into a separate file to make the main one more focused.

## Checklist

- [X] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [X] CI is passing.

 ### Not applicable
 
- [ ] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [ ] The code generator is run and the code is formatted (via `just precommit`).

### CODE REVIEW

File: book/src/SUMMARY.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 - [🧭 Quickstart](quickstart.md)
 - [📚 Tutorial: A Flutter+Rust app](tutorial_with_flutter.md)
+	- [Android setup](tutorial/setup_android.md)
+		- [Alternative NDK setup](tutorial/alternative_ndk.md)
```
Comment (fzyzcjy) [inline book/src/SUMMARY.md:10]:
```suggestion
  - [Android setup](tutorial/setup_android.md)
    - [Alternative NDK setup](tutorial/alternative_ndk.md)
```

**Comment by fzyzcjy** [comment]
Given that the template doc is written by @Desdaemon, I will follow his opinion about this PR :)

**Comment by Desdaemon** [comment]
LGTM

**Comment by fzyzcjy** [comment]
@all-contributors please add @Voklen for doc

**Pull request merged**